lua/core/script.lua

```diff
@@ -111,30 +111,23 @@ end
 --- load a script from the /scripts folder.
 -- @tparam string filename file to load. leave blank to reload current file.
 Script.load = function(filename)
-  local name, path, relative
+  local name, path
   if filename == nil then
     filename = norns.state.script
     name = norns.state.name
     path = norns.state.path
   else
-	if string.sub(filename,1,1) == "/" then
-	  relative = string.sub(filename,string.len(_path["dust"]))
-	else
-	  relative = filename
-	  filename = _path["dust"] .. filename
-	end
-
-	local t = tab.split(string.sub(relative,0,-5),"/")
-	if t[#t] == t[#t-1] then
-	  name = t[#t]
-	else
-	  name = t[#t-1].."/"..t[#t]
-	end
-  if #t==4 then name = t[2].."/"..name end -- dumb hack for 3-deep subfolers
-	path = string.sub(_path["dust"],0,-2)
-	for i = 1,#t-1 do path = path .. "/" .. t[i] end
-	--print("name "..name)
-	--print("final path "..path)
+    filename = string.sub(filename,1,1) == "/" and filename or _path["dust"]..filename
+    path, scriptname = filename:match("^(.*)/([^.]*).*$")
+    name = string.sub(path, string.len(_path["code"]) + 1)
```

Comment:
The `+ 1` assumes that `_path["code"]` doesn't have a trailing slash. If that's an uncomfortable assumption, I can make it check.

Reply:
`_path["code"]` is hard-defined so we're safe